src/odemis/gui/cont/tabs.py

```diff
@@ -3409,6 +3416,10 @@ def __init__(self, name, button, panel, main_frame, main_data):
         if not main_data.ebeamControlsMag:
             main_data.ebeam.magnification.subscribe(self._onSEMMag)
 
+    @wxlimit_invocation(1)
```

Comment:
That might be not frequent enough for the user. I think you can lower it to 0.5s (as the code is really not CPU intensive, it should be completely fine).